src/DynamoDbCache.php

```diff
@@ -582,21 +593,29 @@ private function getKey(string $key): string
      */
     private function getRawItem(string $key): array
     {
-        $input = [
-            'Key' => [
-                $this->primaryField => [
-                    'S' => $key,
-                ],
-            ],
-            'TableName' => $this->tableName,
-        ];
-
         try {
-            $item = $this->client->getItem($input);
+            $item = $this->client->getItem([
+                'Key' => [
+                    $this->primaryField => [
+                        'S' => $key,
+                    ],
+                ],
+                'TableName' => $this->tableName,
+            ]);
 
             return $item->getItem();
         } catch (NetworkException $e) {
-            throw new CacheItemNotFoundException('', 0, $e);
+            switch ($this->networkErrorMode) {
+                case NetworkErrorMode::IGNORE:
+                    throw new CacheItemNotFoundException('', 0, $e);
+                case NetworkErrorMode::THROW:
+                    throw $e;
+                case NetworkErrorMode::WARNING:
+                    trigger_error("Network error when connecting to DynamoDB: {$e->getMessage()}", E_USER_WARNING);
```

Comment:
Another option would be to depend on PSR-3 and log that way. 🤷🏼‍♂️

Reply:
May add that in the future.